### PR TITLE
Enhance prod minification

### PR DESCRIPTION
**Pull request by xD3CODER**
Added postcss-csso for css/scss minification
Added DEV_SERVER_HOST_URL directly in .env (dev server under subdomain support)

### CODE REVIEW

**Comment by calvinl** [comment]
Thanks for this, will review over the weekend.

File: package.json [hunk 1]
```diff
@@ -124,16 +125,19 @@
     "redbox-react": "^1.5.0",
     "supertest": "^3.0.0",
     "webpack-dev-server": "^2.9.6",
```
Comment (calvinl) [inline package.json]: Can you remove this line?

File: package.json [hunk 2]
```diff
@@ -124,16 +125,19 @@
...
+    "webpack-dev-server": "^2.9.7",
     "webpack-sources": "^1.1.0"
   },
   "dependencies": {
     "app-module-path": "^2.2.0",
     "autoprefixer": "^7.2.1",
```
Comment (calvinl) [inline package.json]: This line as well.

File: package.json [hunk 3]
```diff
@@ -124,16 +125,19 @@
...
+    "autoprefixer": "^7.2.2",
     "axios": "^0.17.1",
     "babel": "^6.23.0",
     "babel-cli": "^6.26.0",
     "babel-core": "^6.26.0",
     "babel-loader": "^7.1.1",
```
Comment (calvinl) [inline package.json]: And this. :)

Please also commit package-lock.json

File: postcss.config.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
 const autoprefixer = require('autoprefixer');
+const csso = require('postcss-csso')({restructure: true, comments: false});
```
Comment (calvinl) [inline postcss.config.js]: style should have spacing between curly braces, e.g.
`{restructure: true, comments: false}`
to
`{ restructure: true, comments: false }`

File: webpack/production.babel.js [hunk 5]
```diff
@@ -1,19 +1,39 @@
...
 import CompressionPlugin from 'compression-webpack-plugin';
+import CleanWebpackPlugin from 'clean-webpack-plugin';
+import UglifyJSPlugin from 'uglifyjs-webpack-plugin';
+let pathsToClean = [
+  'dist',
+];
 
+// the clean options to use
+let cleanOptions = {
+  root: '/var/www/react/',
```
Comment (calvinl) [inline webpack/production.babel.js]: Wouldn't `__dirname` be better here?

File: webpack/production.babel.js [hunk 6]
```diff
@@ -1,19 +1,39 @@
 import webpack from 'webpack';
 import baseConfig from './base';
 import CompressionPlugin from 'compression-webpack-plugin';
+import CleanWebpackPlugin from 'clean-webpack-plugin';
```
Comment (calvinl) [inline webpack/production.babel.js]: In production, this package is missing. Please add to `package.json`.

File: server/server.js [hunk 7]
```diff
@@ -33,7 +33,7 @@ app.use(compression());
 
 // Add middleware to serve up all static files
 app.use('/dist',
-  express.static(path.join(__dirname, '../dist')),
+  express.static(path.join(__dirname, '../dist/public')),
```
Comment (calvinl) [inline server/server.js]: let's use `process.env.WEBPACK_OUTPUT_PATH` here instead.

**Review by calvinl — COMMENTED** [review]
Just a few (minor) changes. Looks good otherwise!

**Comment by xD3CODER** [comment]
Yeah i made few mistakes with my PR I update these

**Pull request merged**